src/main/java/organice/logic/commands/FindCommand.java

```diff
@@ -1,44 +1,154 @@
+        ArrayList<Person> tolerablePersons = new ArrayList<>(inputList.size());
+        for (int i = 0; i < inputList.size(); i++) {
+            int levenshteinDistance = distanceList.get(i);
+            if (levenshteinDistance <= FUZZY_THRESHOLD) {
+                distancesOfTolerablePersons.add(levenshteinDistance);
+                tolerablePersons.add(inputList.get(i));
+            }
+        }
 
-    public FindCommand(PersonContainsPrefixesPredicate predicate) {
-        requireNonNull(predicate);
-        this.predicate = predicate;
+        ArrayList<Person> sortedPersons = new ArrayList<>(tolerablePersons);
+        sortedPersons.sort(Comparator.comparingInt(left ->
+                distancesOfTolerablePersons.get(tolerablePersons.indexOf(left))));
+
+        return sortedPersons;
+    }
+
+    private int findMinLevenshteinDistance(List<String> prefixKeywords, String personAttribute) {
+        return prefixKeywords.stream().reduce(Integer.MAX_VALUE, (minDistance, nextKeyword) -> Integer.min(
+                minDistance, calculateLevenshteinDistance(nextKeyword.toLowerCase(), personAttribute.toLowerCase())),
+                Integer::min);
+    }
+
+    /**
+     * Returns the minimum Levenshtein Distance of every {@code prefixKeyword} and {@code personName} pair. If
+     * {@code prefixKeyword} is a single-word string, {@code personName} is split according to spaces (if possible) and
+     * the minimum distance is calculated between every possible pair.
+     */
+    private int findMinLevenshteinDistance(List<String> prefixKeywords, Name personName) {
+        // If keyword is one word long, split the personName and find minLD.
+        // Else just do as we are normally doing in original finalMinLD
+
+        BiFunction<String, String, Integer> findDistanceSplitIfMultiWord = (prefixKeyword, pName) ->
+                prefixKeyword.split(" ").length == 1 ? Arrays.stream(pName.split(" "))
+                        .reduce(Integer.MAX_VALUE, (minDistance, nextNameWord) -> Integer.min(minDistance,
+                                calculateLevenshteinDistance(prefixKeyword.toLowerCase(), nextNameWord.toLowerCase())),
+                                Integer::min)
+                : calculateLevenshteinDistance(prefixKeyword.toLowerCase(), pName.toLowerCase());
+
+        return prefixKeywords.stream().reduce(Integer.MAX_VALUE, (minDistance, nextKeyword) ->
+                Integer.min(minDistance, findDistanceSplitIfMultiWord.apply(nextKeyword, personName.toString())),
+                Integer::min);
     }
 
     @Override
     public CommandResult execute(Model model) {
         requireNonNull(model);
-        model.updateFilteredPersonList(predicate);
-        return new CommandResult(
-                String.format(Messages.MESSAGE_PERSONS_LISTED_OVERVIEW, model.getFilteredPersonList().size()));
+
+        List<Person> allPersons = Arrays.asList(model.getFilteredPersonList().toArray(Person[]::new));
```

Comment:
To get all persons in ORGANice, the initial predicate should be `PREDICATE_SHOW_ALL_PERSONS`.

Reply:
Created new method in `ModelManager` to handle this explicit behaviour.